Mage.Sets/src/mage/cards/s/StrefanMaurerProgenitor.java

```diff
@@ -0,0 +1,215 @@
+                new BeginningOfEndStepTriggeredAbility(
+                        new CreateBloodTokenEffect(),
+                        TargetController.YOU,
+                        false)
+                        .addHint(hint),
+                new PlayersWhoLifeLostWatcher()
+        );
+
+        // Whenever Strefan attacks, you may sacrifice two Blood tokens.
+        // If you do, you may put a Vampire card from your hand onto the battlefield tapped and attacking.
+        // It gains indestructible until end of turn.
+        this.addAbility(new AttacksTriggeredAbility(
+                new DoIfCostPaid(
+                        new PlayVampireEffect(),
+                        new SacrificeTargetCost(new TargetControlledPermanent(
+                                2,
+                                2,
+                                bloodTokenFilter,
+                                true)
+                        )),
+                false,
+                "When Strefan attacks, "
+        ));
+    }
+
+    private StrefanMaurerProgenitor(final StrefanMaurerProgenitor card) { super(card);}
+
+    @Override
+    public Card copy() {
+        return new StrefanMaurerProgenitor(this);
+    }
+}
+
+class PlayVampireEffect extends OneShotEffect {
+
+    private static final FilterCreatureCard vampireCardFilter = new FilterCreatureCard();
+    static { vampireCardFilter.add(SubType.VAMPIRE.getPredicate()); }
+
+    public PlayVampireEffect() {
+        super(Outcome.PutCreatureInPlay);
+        staticText = "you may put a Vampire card from your hand onto the battlefield tapped and attacking. " +
+                "It gains indestructible until end of turn.";
+    }
+
+    private PlayVampireEffect(final PlayVampireEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) { return false; }
+
+        TargetCard target = new TargetCardInHand(0, 1, vampireCardFilter);
+        if (!player.choose(outcome, player.getHand(), target, game)) { return false; }
+
+        Card card = game.getCard(target.getFirstTarget());
+        if (card == null) { return false; }
+
+        player.moveCards(card, Zone.BATTLEFIELD, source, game, true, false, true, null);
+
+        Permanent permanent = game.getPermanent(card.getId());
+        if (permanent == null) { return false; }
+
+        game.getCombat().addAttackingCreature(permanent.getId(), game);
+
+        // Gains indestructable until end of turn
+        ContinuousEffect effect = new GainAbilityTargetEffect(IndestructibleAbility.getInstance(), Duration.EndOfTurn);
+        effect.setTargetPointer(new FixedTarget(permanent.getId(), game));
+        game.addEffect(effect, source);
+
+        return true;
+    }
+
+    @Override
+    public PlayVampireEffect copy() { return new PlayVampireEffect(this); }
+}
+
+class CreateBloodTokenEffect extends OneShotEffect {
```

Comment:
This doesn't need to be a custom class.  `CreateTokenEffect` can take in a DynamicValue (which you've already written)